base/abstractarray.jl

```diff
@@ -2009,7 +2009,7 @@ concatenate_setindex!(R, X::AbstractArray, I...) = (R[I...] = X)
 
 function map!(f::F, dest::AbstractArray, A::AbstractArray) where F
     for (i,j) in zip(eachindex(dest),eachindex(A))
-        dest[i] = f(A[j])
+        @inbounds dest[i] = f(A[j])
```

Comment:
I don't think `f()` is inbounds-guaranteed here. For example, if it was `i -> [][i]`, that would be always OOB.
```suggestion
        val = f(@inbounds A[j])
        @inbounds dest[i] = val
```